pkg/controller/tidbcluster/pod_control.go

```diff
@@ -249,6 +249,10 @@ func (c *PodController) syncPDPod(ctx context.Context, pod *corev1.Pod, tc *v1al
 		return reconcile.Result{}, nil
 	}
 
+	if c.isEvictLeaderExpired(pod, v1alpha1.PDLeaderTransferExpirationTimeAnnKey) {
+		return reconcile.Result{}, c.cleanupLeaderEvictionAnnotations(pod, tc, []string{v1alpha1.PDLeaderTransferAnnKey, v1alpha1.PDLeaderTransferExpirationTimeAnnKey})
```

Comment:
It seems the annotation is cleared during pod sync if the expiration time is reached. Would it be a problem if next time when the operator enters this reconciliation loop(well at that moment it's still expired), the `isEvictLeaderExpired` would return `false` since the annotation is removed, and then it proceeds with the leader eviction?

Reply:
Ok, that makes sense.